File: python/writer/generators/test_material_pbr.py
# SPDX-License-Identifier: MIT
from pathlib import Path

from .common import run


def generate(skppy, destination: Path) -> None:
    model = skppy.new_model()
    model.add_material(
        "PBRMaterial",
        color=skppy.Color(180, 180, 200),
        metallic=0.8,
        roughness=0.25,
    )
    skppy.save(model, destination)


if __name__ == "__main__":
    raise SystemExit(run(generate, "material_pbr"))

File: c/generators/test_material_pbr.c
// SPDX-License-Identifier: MIT
#include "test_utils.h"
#include <SketchUpAPI/model/model.h>
#include <SketchUpAPI/model/material.h>
#include <SketchUpAPI/color.h>
#include <stdio.h>

int main(int argc, char** argv) {
    if (!init_sketchup()) return 1;
    enum SUModelVersion ver = version_from_args(argc, argv);

    SUModelRef model = SU_INVALID;
    SUModelCreate(&model);

    // Create a PBR material with explicit metallic / roughness factors.
    SUMaterialRef material = SU_INVALID;
    SUMaterialCreate(&material);
    SUMaterialSetName(material, "PBRMaterial");

    SUColor base = { 180, 180, 200, 255 };
    SUMaterialSetColor(material, &base);

    // Enable and set PBR metallic / roughness factors.
    SUMaterialSetMetalnessEnabled(material, true);
    SUMaterialSetMetallicFactor(material, 0.8);

    SUMaterialSetRoughnessEnabled(material, true);
    SUMaterialSetRoughnessFactor(material, 0.25);

    SUModelAddMaterials(model, 1, &material);

    save_model_with_version(model, "data/material_pbr.skp", ver);

    SUModelRelease(&model);
    term_sketchup();
    return 0;
}
